src/shogun/lib/any.h

```diff
@@ -212,6 +283,105 @@ namespace shogun
             policy->clear(&storage);
         }
 
+		template <class Archive, class Type>
+		void cereal_save_helper(Archive& ar) const
+		{
+			ar(*(reinterpret_cast<Type*>(storage)));
+		}
+
+		template<class Archive>
+		void cereal_save(Archive& ar) const
+		{
+			ar (m_datatype.e_containertype);
+			ar (m_datatype.e_primitivetype);
+			switch (m_datatype.e_containertype) {
+			case serial::EnumContainerType::CT_PRIMITIVE:
+				switch (m_datatype.e_primitivetype) {
+				case serial::EnumPrimitiveType::PT_INT_32:
+					cereal_save_helper<Archive, int32_t>(ar);
+					break;
+				case serial::EnumPrimitiveType::PT_FLOAT_64:
+					cereal_save_helper<Archive, float64_t>(ar);
+					break;
+				case serial::EnumPrimitiveType::PT_UNDEFINED:
+					SG_SERROR("Type error: undefined data type cannot be serialized.\n");
+					break;
+				}
+				break;
+			case serial::EnumContainerType::CT_SGVECTOR:
+				switch (m_datatype.e_primitivetype) {
+				case serial::EnumPrimitiveType::PT_INT_32:
+					cereal_save_helper<Archive, SGVector<int32_t> >(ar);
+					break;
+				case serial::EnumPrimitiveType::PT_FLOAT_64:
```

Comment:
BTW it would be cool if the whole serailization framework would have debug message that are very clear to read. They should include the parameter name, the type, etc